### PR TITLE
Removed old PHP version checks

### PR DESCRIPTION
**Pull request by webimpress**
Repository supports PHP 5.4+

### CODE REVIEW

File: src/Runner.php [hunk 1]
```diff
@@ -229,12 +229,6 @@ public function runPHPCBF()
      */
     public function checkRequirements()
     {
-        // Check the PHP version.
-        if (PHP_VERSION_ID < 50400) {
-            $error = 'ERROR: PHP_CodeSniffer requires PHP version 5.4.0 or greater.'.PHP_EOL;
```
Comment (jrfnl) [inline src/Runner.php]: This one looks like it should not be removed. This is the error which will be thrown if for whatever reason someone run PHPCS 3.x on PHP 5.3.

File: src/Standards/Generic/Tests/PHP/DisallowShortOpenTagUnitTest.php [hunk 2]
```diff
@@ -52,13 +52,6 @@ public function getErrorList($testFile='')
     {
         switch ($testFile) {
         case 'DisallowShortOpenTagUnitTest.1.inc':
-            if (PHP_VERSION_ID < 50400) {
```
Comment (jrfnl) [inline src/Standards/Generic/Tests/PHP/DisallowShortOpenTagUnitTest.php]: I think more changes are needed for the DisallowShortOpenTag sniff. I'll have a look.

File: src/Runner.php (hunk 1 above)
Comment (webimpress) [reply src/Runner.php]: The requirement is set in composer and I think it should be enough.

File: src/Standards/Generic/Tests/PHP/DisallowShortOpenTagUnitTest.php (hunk 2 above)
Comment (webimpress) [reply src/Standards/Generic/Tests/PHP/DisallowShortOpenTagUnitTest.php]: @jrfnl Did you have chance to check it? I don't know what do you mean...

File: src/Runner.php (hunk 1 above)
Comment (jrfnl) [reply src/Runner.php]: @webimpress By far not all people use PHPCS via Composer, so I don't necessarily agree.

File: src/Standards/Generic/Tests/PHP/DisallowShortOpenTagUnitTest.php (hunk 2 above)
Comment (jrfnl) [reply src/Standards/Generic/Tests/PHP/DisallowShortOpenTagUnitTest.php]: @webimpress Sorry, I'm at a conference at the moment, I'll try and get to it over the next few days.

File: src/Runner.php (hunk 1 above)
Comment (gsherwood) [reply src/Runner.php]: This check needs to stay.

Comment (webimpress) [reply src/Runner.php]: Ok then, should I revert it or you will do it on merge?

Comment (gsherwood) [reply src/Runner.php]: I've just noticed the autoloader fails for < 5.4, so leave it. I think I need to move the requirement check higher up.

File: src/Standards/Generic/Tests/PHP/DisallowShortOpenTagUnitTest.php (hunk 2 above)
Comment (jrfnl) [reply src/Standards/Generic/Tests/PHP/DisallowShortOpenTagUnitTest.php]: Ok. Finally had a chance to look at this properly. I was confusing this sniff with the sister-sniff `DisallowAlternativePHPTags`. I'm sending you a PR with a very minor addition to this PR.

Other than that, I noticed another issue regarding the `DisallowShortOpenTag` sniff (and it's sister), but that should be handled in a separate PR - see #1784.

And another issue regarding the testing of the `DisallowAlternativePHPTags` sniff, but, again, that should be handled in another PR - see #1786.

**Comment by TomasVotruba** [comment]
:+1:

**Comment by gsherwood** [comment]
I ended up just putting the version check back into `checkRequirements()` because I have up trying to get a PHP 5.3 install running again to test/fix the autoloader. Having that version check there will at least remind me to update it if I change the min version again.

Thanks for finding and fixing the other checks.

**Pull request merged**